manifests/init.pp

```diff
@@ -31,14 +31,30 @@
         if ! $gpgkeyfingerprint {
           fail('you must specify a `$gpgkeyfingerprint` when using `$manage_repo` on debian')
         }
+        # place the key in the keyrings directory where apt won't search for keys for all repos
+        file { '/usr/share/keyrings/lldpd.gpg':
+          source => "http://download.opensuse.org/repositories/home:/vbernat/${repourl}/Release.key",
+          owner  => 'root',
+          group  => 'root',
+          mode   => '0644',
+        }
+        # purge old key files that we installed in previous releases
+        $old_keys = ['/etc/apt/trusted.gpg.d/home_vbernat.gpg', '/etc/apt/trusted.gpg.d/home_vbernat.gpg~',]
+        $old_keys.each |Stdlib::Absolutepath $key| {
+          file { $key:
+            ensure => 'absent',
+          }
+        }
+        # previously managed by apt::key, we need to purge it from the global keyring in /etc/apt/trusted.gpg
+        apt::key { 'EF795E4D26E48F1D7661267B431C37A97C3E114B':
```

Comment:
The CI warnings before the failure makes me think we should `include apt` before this line and the error would go away.

Reply:
fixed it and extended the tests